Move firewall doc under "Application" section (#534)
NethServer/docs: administrator-manual/en/firewall.rst

.. _firewall-section:

========
Firewall
========

|product| can act as :index:`firewall` and :index:`gateway` inside the network where it is installed.
All traffic between computers on the local network and the Internet passes through the server that decides how to 
route packets and what rules to apply.
Firewall mode is enabled only if the system has at least one network interface configured with red role.

The Firewall application can be installed from :ref:`software-center-section` and includes:
 
* Multi WAN support up to 15 connections
* Firewall rules management
* Traffic shaping (QoS)
* Port forwarding
* Routing rules to divert traffic on a specific WAN

> * Intrusion Prevention System (IPS)

Author: ```suggestion
```

* Deep packet inspection (DPI)
* Smart search to quickly find existing rules or objects
* Real time charts

Real time charts display traffic and service statistics collected by `netdata <https://www.netdata.cloud/>`_.
To avoid performance penalty on slow hardware, netdata is not part of the firewall application and can be installed from :ref:`software-center-section`.

.. _apply_revert-section:

Apply and revert
================

Every time the firewall configuration has been changed, modifications are not applied immediately but saved in a temporary store.
For the changes to take effect, click on the :guilabel:`Apply` button at the top right corner of the page.

> As long as the new rules created have not been applied, you can revert all changes clicking the :guilabel:`Revert` button at the top right corner of the page.

Reviewer: ```suggestion
As long as the new rules created have not been applied, you can revert all changes by clicking the :guilabel:`Revert` button at the top right corner of the page.
```

traffic should be blocked or allowed. 
:dfn:`Policies` are the default rules to be applied when the network traffic does not match any existing criteria.

The firewall implements two default policies:

* :dfn:`Allowed`: all traffic from green to red is allowed
* :dfn:`Blocked`: all traffic from green to red network is blocked. Specific traffic must be allowed with custom rules.

To change the default policy for Internet access, enable or disable the :guilabel:`Traffic to Internet (red interface)` option from the :guilabel:`Settings` page.

> Policies can be changed also by creating specific rules between zones from :guilabel:`Rules` page.

Reviewer: ```suggestion
Policies can be changed also by creating specific rules between zones from the :guilabel:`Rules` page.
```

Author: `the` it's not really needed here, but ok

* Destination: Server1 
* Service: -

.. _wan-section:

WAN
===

The term :dfn:`WAN` (Wide Area Network) refers to a public network outside the server, usually connected to the Internet. 
A :dfn:`provider` is the company that actually manages the :index:`WAN` link.

All WAN network interfaces are labeled with the red role and are listed on the top of the page, just below bandwidth usage charts.
Rules can be created under the :guilabel:`Rules` section on the same page.

If the server has two or more configured red interfaces, it is required to correctly fill, 
:guilabel:`Download bandwidth` and :guilabel:`Upload bandwidth` fields from the :guilabel:`Network` page.

> Download and upload bandwidth can be automatically calculated using :guilabel:`Speedtest` button.

Reviewer: ```suggestion
Download and upload bandwidth can be automatically calculated using the :guilabel:`Speedtest` button.
```

Each provider represents a WAN connection and is associated with a network adapter. 

> Each provider defines a :dfn:`weight`: to an higher the :index:`weight` correspond an higher priority of the network card associated with the provider. 

Reviewer: ```suggestion
Each provider defines a :dfn:`weight`: the higher the weight, the higher the priority of the network card associated with the provider. 
```

Author: I do not like the old sentence.

Reviewer: There's a mistake in the new sentence too: 

```diff
- to an higher the :index:`weight`
+ to an higher :index:`weight`
```

* Number of consecutive lost packets
* Interval in seconds between sent packets

> WAN mode and link monitoring options click on :guilabel:`Configure` button.

Reviewer: I don't understand the sentence

Author: Because it's wrong :(

```suggestion
To change WAN mode and link monitoring options click on :guilabel:`Configure` button.
```

The network traffic can be routed to specific WANs by creating rules inside the :guilabel:`Rules` section on this page.

> After creating or editing rules, make sure to :ref:`apply <apply_revert-section>` the changes.

Reviewer: ```suggestion
After creating or editing rules, make sure to apply the changes. See :ref:`<apply_revert-section>` for details.
```

Limiting access
---------------

By default, the field access to port forward is granted to anyone.
You can restrict access to port forward only from some IP addresses or networks by adding entries to :guilabel:`Restrict access to` field.
This configuration is useful when services should be available only from trusted IPs or networks. 

Example of valid entries:

* ``10.2.10.4``: enable port forward for traffic coming from 10.2.10.4 IP
* ``10.2.10.0/24``: enable port forward only for traffic coming from 10.2.10.0/24 network

.. _snat-section:

> sNAT 1:1

Reviewer: ```suggestion
SNAT 1:1
```

Author: I'm not sure about this ... but ok

========

> One-to-one source NAT is a way to make systems behind a firewall and configured with private IP addresses appear to have public IP addresses.

Reviewer: ```suggestion
One-to-one :index:`source NAT` (:index:`SNAT`) is a way to make systems behind a firewall and configured with private IP addresses appear to have public IP addresses.
```

If you have a bunch of public IP addresses and if you want to associate one of these to a specific network host, :index:`NAT 1:1` is the way.
SNAT is available only if there is at least one IP alias configured on red network interfaces.

This feature only applies to network traffic from a host inside the local network to the public Internet.

> It doesn't affect in any way the traffic from internet toward the alias IP, if you need to route some specific traffic to the internal host use the port forward as usual.

Reviewer: ```suggestion
It does not affect in any way the traffic from the Internet toward the alias IP. If you need to route some specific traffic to the internal host use the port forward as usual.
```

A class will exceed its committed rate only if there is spare bandwidth available.

Traffic shaping classes can be defined under :guilabel:`Traffic shaping` page.
When creating a new class, fill the following fields.

* :guilabel:`Class name`: a representative name
* :guilabel:`Description`: optional description for the class

Limits under :guilabel:`Download bandwidth limits` section:

  * :guilabel:`Min`: minimum reserved download bandwidth, if empty no download reservation will be created
  * :guilabel:`Max`: maximum allowed download bandwidth, if empty no upper limit will be set

Limits under :guilabel:`Upload bandwidth limits` section:

  * :guilabel:`Min`:  minimum reserved upload bandwidth, if empty no upload reservation will be created
  * :guilabel:`Max`: maximum allowed download bandwidth, if empty no upper limit will be created

For each class the bandwidth can be specified using the percentage of available network bandwidth or

> with absolutes values expressed in kbps.

Reviewer: ```suggestion
with absolutes values expressed in Kbps.
```

Author: The right spelling is with lower case: https://en.wikipedia.org/wiki/Kilobit
